synapse/storage/util/id_generators.py

```diff
@@ -157,12 +188,10 @@ def __init__(
         # The key and values are the same, but we never look at the values.
         self._unfinished_ids: OrderedDict[int, int] = OrderedDict()
 
+    def advance(self, instance_name: str, new_id: int) -> None:
+        raise Exception("Replication is not supported by StreamIdGenerator")
+
     def get_next(self) -> AsyncContextManager[int]:
-        """
-        Usage:
-            async with stream_id_gen.get_next() as stream_id:
-                # ... persist event ...
-        """
```

Comment:
I'd argue that these bits of docs shouldn't be removed, i'll note them in the next few comments.

Reply:
They've been moved into `AbstractStreamId{Generator,Tracker}`. I was under the impression that IDEs would pick it up automatically. [inspect.getdoc](https://docs.python.org/3/library/inspect.html#inspect.getdoc) is noted to do the right thing.

Reply:
Ah, interesting, though i'd like a confirmation from a synapse member on this change, because im not 100% sure myself

Reply:
I’ll be closing the other comments, apologies if it came over as overwhelming, I simply meant to highlight all places where this issued occurred without much duplication

Reply:
Not a problem!

Reply:
I wasn't feeling up for making sense of `IdGenerator` and unifying it with the stream id generators. They do look vaguely similar.

Reply:
Turns out `IdGenerator` is far simpler than `StreamIdGenerator`. The former only issues IDs while the latter has to track when they are committed.